### PR TITLE
Allow tab content to be opened using middle click

### PR DESCRIPTION
**Pull request by AdrienClairembault**
### Issue

Opening an item tab using the middle click lead to a broken "raw" page:

image

This is because our tabs `href` properties are the ajax endpoinst that will fetch the data.
So, opening the link in a new tab will only fetch the tab content and nothing else.

### Fix

This is easy to fix as we have no requirement to use this url directly in the `href` property.
We can use a valid `href` and move the ajax endpoints URL into dedicated data attributes.

### Tests

2E tests can't be added to validate this as cypress does not support multiple browsers tabs.

### Branch

I'm doing this changes in main because the `href` property may possibly be parsed by some plugins which would break with this change.
If you think this possible issue is too "far fetched", I can move the PR to target the bugfix branch instead.

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !33040

### CODE REVIEW

**Comment by cconard96** [comment]
Full page tabs are an old debug feature. Not sure they serve much of a purpose anymore and the warning message doesn't seem to trigger anymore.
See `ajax/common.tabs.php`.
Maybe it can be removed altogether.

**Comment by cedric-anne** [comment]
> Full page tabs are an old debug feature. Not sure they serve much of a purpose anymore and the warning message doesn't seem to trigger anymore. See `ajax/common.tabs.php`. Maybe it can be removed altogether.

Done in #17157 .

File: src/Ajax.php [hunk 1]
```diff
@@ -344,15 +344,33 @@ public static function createTabs(
...
+                $tab_content_url = $val['url'] . (isset($val['params']) ? '?' . $val['params'] : '');
                 $selected = $active_id == $target ? 'selected' : '';
                 $title = $val['title'];
                 $title_clean = strip_tags($title);
+
+                // Compute direct link that user can reach in a new tab using
+                // middle mouse click.
+                // TODO: ctrl+click should have the same behavior but it seems
+                // to be caught by the tabs events handler and does not trigger
+                // a new browser tab.
```
Comment (cedric-anne) [inline src/Ajax.php:356]: Clicks default behaviour are prevented by the bootstrap tabs logic. The only way to handle this is to remove the `data-bs-toggle='tab'` attribute and add a click handler that either let the default behaviour continue when `e.ctrlKey === true`, either trigger the tab display manually.

Comment (AdrienClairembault) [reply src/Ajax.php:356]: Yeah, I am not sure it is worth dealing with right now as it doesn't break anything.

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Pull request merged**